packages/checkbox/src/index.mts

```diff
@@ -4,154 +4,86 @@ import {
   useKeypress,
   usePrefix,
   usePagination,
-  isUpKey,
-  isDownKey,
+  useScroll,
+  useSpeedDial,
   isSpaceKey,
   isNumberKey,
   isEnterKey,
   Separator,
+  useRef,
 } from '@inquirer/core';
 import type {} from '@inquirer/type';
 import chalk from 'chalk';
-import figures from 'figures';
 import ansiEscapes from 'ansi-escapes';
-
-export type Choice<Value> = {
-  name?: string;
-  value: Value;
-  disabled?: boolean | string;
-  checked?: boolean;
-  type?: never;
-};
+import { render } from './render.mjs';
+import { selectable, Item, Choice, toggle, check } from './choice.mjs';
 
 type Config<Value> = {
   prefix?: string;
   pageSize?: number;
   instructions?: string | boolean;
   message: string;
   choices: ReadonlyArray<Choice<Value> | Separator>;
+  loop?: boolean;
 };
 
-function isSelectableChoice<T>(
-  choice: undefined | Separator | Choice<T>,
-): choice is Choice<T> {
-  return choice != null && !Separator.isSeparator(choice) && !choice.disabled;
-}
-
 export default createPrompt(
   <Value extends unknown>(
     config: Config<Value>,
     done: (value: Array<Value>) => void,
   ): string => {
-    const { prefix = usePrefix(), instructions } = config;
-
+    const { prefix = usePrefix(), instructions, pageSize, loop, choices } = config;
+    const firstRender = useRef(true);
     const [status, setStatus] = useState('pending');
-    const [choices, setChoices] = useState<Array<Separator | Choice<Value>>>(() =>
-      config.choices.map((choice) => ({ ...choice })),
+    const [items, setItems] = useState<ReadonlyArray<Item<Value>>>(
+      choices.map((choice) => ({ ...choice })),
     );
-    const [cursorPosition, setCursorPosition] = useState(0);
     const [showHelpTip, setShowHelpTip] = useState(true);
+    const message = chalk.bold(config.message);
+    const { contents, active, setActive } = usePagination<Item<Value>>({
```

Comment:
Can we bring back `useState` at the component level?

And can we go back on the introduction of `useSpeedDial` & `useScroll` (e.g keeping the keypress handler in each prompt.) Those 2 hooks are adding a fairly large diff, and I'm not convinced it's a positive change for the codebase. It's DRYing stuff, but introducing complexity that doesn't feel necessary.